test/facets/thirdparty/accuse.spec.ts

```diff
@@ -0,0 +1,716 @@
+          )
+        );
+      // verify accused archaeologists cursed bonds have zeroed and free bonds haven't been increased
+      accusedArchaeologistProfiles.forEach(
+        (accusedArchaeologistProfile, index) => {
+          expect(accusedArchaeologistProfile.cursedBond.toString()).to.equal(
+            "0"
+          );
+          // archaeologist's free bond prior to accusal
+          const initialFreeBond =
+            archaeologistAddressesToInitialFreeBondsSarquitos.get(
+              accusedArchaeologists[index].archAddress
+            );
+          expect(accusedArchaeologistProfile.freeBond.toString()).to.equal(
+            initialFreeBond
+          );
+        }
+      );
+
+      // verify innocent archaeologist bond has been freed - cursed bond set to zero and free bond increased by digging fee (cursed bond amount)
+      innocentArchaeologistProfiles.forEach(
+        (innocentArchaeologistProfile, index) => {
+          expect(innocentArchaeologistProfile.cursedBond.toString()).to.equal(
+            "0"
+          );
+          // archaeologist's free bond prior to accusal
+          const initialFreeBond =
+            archaeologistAddressesToInitialFreeBondsSarquitos.get(
+              innocentArchaeologists[index].archAddress
+            );
+          expect(innocentArchaeologistProfile.freeBond.toString()).to.equal(
+            initialFreeBond!.add(innocentArchaeologists[index].diggingFee)
+          );
+        }
+      );
+
+      // verify accused archaeologists have been marked as accused
+      await verifyAccusalStatusesForArchaeologists(
+        sarcophagus.sarcoId,
+        accusedArchaeologists,
+        true
+      );
+
+      // verify innocent archaeologists have not been marked as accused
+      await verifyAccusalStatusesForArchaeologists(
+        sarcophagus.sarcoId,
+        innocentArchaeologists,
+        false
+      );
+
+      // verify the sarcoId has been added to the accused archaeologist's archaeologistAccusals
+      await Promise.all(
+        accusedArchaeologists.map(async (accusedArchaeologist) => {
+          const accusedArchaeologistAccusalsCount =
+            await viewStateFacet.getArchaeologistAccusalsCount(
+              accusedArchaeologist.archAddress
+            );
+          expect(accusedArchaeologistAccusalsCount.toString()).to.equal("1");
+        })
+      );
+
+      // verify the sarcoId has not been added to the innocent archaeologist's archaeologistAccusals
+      await Promise.all(
+        innocentArchaeologists.map(async (innocentArchaeologist) => {
+          const innocentArchaeologistAccusalsCount =
+            await viewStateFacet.getArchaeologistAccusalsCount(
+              innocentArchaeologist.archAddress
+            );
+          expect(innocentArchaeologistAccusalsCount.toString()).to.equal("0");
+        })
+      );
+    });
+  });
+
+  // todo: can an async context be used with mocha?
```

Comment:
I don't think that this is something that can actually be done with mocha, but if anybody has suggestions on better ways to organize a set of tests where there's some asynchronous configuration and then several tests that use that configuration I'd like to hear them. Currently, there's a lot of duplication across tests because I'm having to set up the sarcophagus and accusal call over and over again.

Below is an example of what I'd like to be able to write, but `context` is an alias for `describe` which cannot take an async callback. We have the option to run mocha with `--delay` to do some asynchronous configuration before running the suite but I don't think it gives us the organizational advantage that context/describe do.

Reply:
`beforeEach`, maybe?

Reply:
I think this is what Kelvin is suggesting 
```
let sarcohpagus: Sarcophagus;
...

beforeEach(async () => {
  sarcohpagus = await getSarcophagus();
  ...
})

it("should do stuff", () => {
  // use sarcophagus var
})
```

Reply:
What you're describing has always been a pain point for me that has made me rip what little hair I have left out of my head. I still haven't come to a satisfactory answer for this, but extracting common code into fixtures is the way I usually go. This is what the current test fixtures are supposed to do, but I'm not sure if they've retained their original purpose over time.